cpp/src/arrow/filesystem/mockfs.cc

```diff
@@ -685,9 +685,7 @@ Status MockFileSystem::Make(TimePoint current_time, const std::vector<FileStats>
     }
   }
 
-  *out = fs;
-
-  return Status::OK();
+  return fs;
```

Comment:
This will need a move on some platforms to avoid a spurious copy (and clang will sometimes emit a warning about it)
```suggestion
  return std::move(fs);
```

Reply:
Every time I do this, I get a warning that it may disable RVO. Anyway, this is a mock class.